programs/zstdcli.c

```diff
@@ -1480,24 +1480,27 @@ int main(int argCount, const char* argv[])
 
     if (showDefaultCParams) {
         if (operation == zom_decompress) {
-            DISPLAY("error : can't use --show-default-cparams in decompression mode \n");
+            DISPLAYLEVEL(1, "error : can't use --show-default-cparams in decompression mode \n");
             CLEAN_RETURN(1);
         }
     }
 
     if (dictFileName != NULL && patchFromDictFileName != NULL) {
-        DISPLAY("error : can't use -D and --patch-from=# at the same time \n");
+        DISPLAYLEVEL(1, "error : can't use -D and --patch-from=# at the same time \n");
         CLEAN_RETURN(1);
     }
 
     if (patchFromDictFileName != NULL && filenames->tableSize > 1) {
-        DISPLAY("error : can't use --patch-from=# on multiple files \n");
+        DISPLAYLEVEL(1, "error : can't use --patch-from=# on multiple files \n");
         CLEAN_RETURN(1);
     }
 
-    /* No status message in pipe mode (stdin - stdout) */
+    /* No status message by default when output is stdout */
     hasStdout = outFileName && !strcmp(outFileName,stdoutmark);
-    if ((hasStdout || !UTIL_isConsole(stderr)) && (g_displayLevel==2)) g_displayLevel=1;
+    if (hasStdout && (g_displayLevel==2)) g_displayLevel=1;
+
+    /* when stderr is not the console, do not pollute it with status updates */
+    if (!UTIL_isConsole(stderr)) FIO_setProgressSetting(FIO_ps_never);
```

Comment:
This doesn't quite work, because this breaks the `--progress` flag. When stderr is not a console, `--progress` won't work. Which is probably not the intended behavior.

Reply:
I think you basically want:

```c
if (!UTIL_isConsole(stderr) && FIO_getProgressSetting() == FIO_ps_auto) FIO_setProgressSetting(FIO_ps_never);
```

Though you'll probably have to add the function `FIO_getProgressSetting()`.